hie-plugin-api/Haskell/Ide/Engine/Cradle.hs

```diff
@@ -14,8 +32,296 @@ import HIE.Bios as BIOS
+                   , cradleOptsProg =
+                       CradleAction { actionName = "Cabal-Helper-"
+                                        ++ actionNameSuffix
+                                        ++ "-None"
+                                    , runCradle = \_ -> return CradleNone
+                                    }
+                   }
+        Just realPackage -> do
+          debugm $ "Cabal-Helper cradle package: " ++ show realPackage
+          -- Field `pSourceDir` often has the form `<cwd>/./plugin`
+          -- but we only want `<cwd>/plugin`
+          let normalisedPackageLocation = normalise $ pSourceDir realPackage
+          debugm
+            $ "Cabal-Helper normalisedPackageLocation: "
+            ++ normalisedPackageLocation
+          return
+            Cradle { cradleRootDir = normalisedPackageLocation
+                   , cradleOptsProg =
+                       CradleAction { actionName =
+                                        "Cabal-Helper-" ++ actionNameSuffix
+                                    , runCradle = cabalHelperAction
+                                        env
+                                        realPackage
+                                        normalisedPackageLocation
+                                    }
+                   }
+    where
+      -- | Cradle Action to query for the ComponentOptions that are needed
+      -- to load the given FilePath.
+      -- This Function is not supposed to throw any exceptions and use
+      -- 'CradleLoadResult' to indicate errors.
+      cabalHelperAction :: QueryEnv v -- ^ Query Env created by 'mkQueryEnv'
+                                      -- with the appropriate 'distdir'
+                        -> Package v -- ^ Package this Cradle is part for.
+                        -> FilePath -- ^ Root directory of the cradle
+                                    -- this action belongs to.
+                        -> FilePath -- ^ FilePath to load, expected to be an absolute path.
+                        -> IO (CradleLoadResult ComponentOptions)
+      cabalHelperAction env package root fp = do
+        -- Get all unit infos the given FilePath may belong to
+        let units = pUnits package
+        -- make the FilePath to load relative to the root of the cradle.
+        let relativeFp = makeRelative root fp
+        debugm $ "Relative Module FilePath: " ++ relativeFp
+        getComponent env (toList units) relativeFp
+          >>= \case
+            Just comp -> do
+              let fs = getFlags comp
+              let targets = getTargets comp relativeFp
+              let ghcOptions = fs ++ targets
+              debugm $ "Flags for \"" ++ fp ++ "\": " ++ show ghcOptions
+              debugm $ "Component Infos: " ++ show comp
+              return
+                $ CradleSuccess
+                  ComponentOptions { componentOptions = ghcOptions
+                                   , componentDependencies = []
```

Comment:
No dependencies?

Reply:
Not right now, since HIE cant work with it anyways. In the end, this would be a reimplementation of the code in hie-bios.